main.js

```diff
@@ -83,9 +84,11 @@ function showMainScreen() {
 
 function keyPressed() {
 	// Allow the user to reset the game via a special button
-	if (key === 'F2') state = MAIN_SCREEN_STATE
+	if (key === 'F2') {
+    state = MAIN_SCREEN_STATE
+    currentMode.onReturnToMenu()
```

Comment:
Why do you have to call this?  You already call this inside the `ConfigurationScreen` upon return.  You're assuming currentMode is a particular class/type for this call to even be valid.

Reply:
This is being called because currently when F2 is pressed it returns to the main menu screen but does not hide the file selector and input elements.